### PR TITLE
HCLS Integration Test

### PR DESCRIPTION
**Pull request by cdunbar13**
Initial test for HCLS.  Creates cluster, runs some basic mount and partitions tests.

Had to make updates to slurm-integration-test so it would run multigroup deployments.  It should still be compatible with the other integration tests since it just goes from terraform commands to ghpc.  

Added a login IP check because without `disable_login_public_ips: false` the integration test can fail without triggering the rescue or the always block in the integration test section (no hosts so it skips the integration tests without running always).

### CODE REVIEW

File: tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml
Comment (nick-stroud) [inline tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml:1]: How is this different from `rescue_terraform_failure.yml`, should we be instead renaming the file (instead of copying) and moving over all usages?

File: docs/videos/healthcare-and-life-sciences/hcls-blueprint.yaml [hunk 1]
```diff
@@ -48,9 +48,6 @@ deployment_groups:
 
   - id: network1
     source: modules/network/vpc
-    settings:
-      network_name: hcls-cluster-net
-      subnetwork_name: primary-subnet
```
Comment (nick-stroud) [inline docs/videos/healthcare-and-life-sciences/hcls-blueprint.yaml:53]: Why was this needed?

Comment (cdunbar13) [reply docs/videos/healthcare-and-life-sciences/hcls-blueprint.yaml:53]: This was needed because if anything were to break and the was left behind, all other deployments would fail because this VPC is hardcoded.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

**Comment by nick-stroud** [comment]
I am fine to merge over the current failing tests (batch-mpi: known unrelated failure, and centos7: capacity), but please see comments.

File: tools/cloud-build/daily-tests/ansible_playbooks/slurm-integration-test.yml [hunk 2]
```diff
@@ -89,7 +84,10 @@
       ansible.builtin.set_fact:
         login_ip: "{{ get_login_ip.stdout }}"
       when: '"*" in login_node'
-
+    - name: Check that login IP is set
+      ansible.builtin.assert:
+        that:
+        - login_ip != ""
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply tools/cloud-build/daily-tests/ansible_playbooks/slurm-integration-test.yml:90]: I'm new to Ansible, so I'm not 100% sure, but I think an empty string is different than a undefined variable.

File: tools/cloud-build/daily-tests/tests/hcls.yml
[the remark this answers is not in the document]
Comment (cdunbar13) [reply tools/cloud-build/daily-tests/tests/hcls.yml:1]: Yes, I have had several successful runs of it (not through triggering, but from manually running it with `gcloud builds ...`

File: tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml [hunk 3]
```diff
@@ -30,11 +30,12 @@
     - delete
     - "{{ deployment_name }}"
 
-- name: Tear Down Cluster
-  changed_when: true  # assume something destroyed
-  run_once: true
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml:35]: I took this code from another test, so I'm assuming it's functional (besides having tested it).  It seems to now be able to track ghpc so we can get rid of the assumption of something being changed.  Additionally the `run_once` is redundant (at least in the case) because it's delegated to localhost in `slurm-integration-test.yml`.  That being said I've re-added it in just to make sure if any other script uses it, it'll be handled correctly.

File: tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml [hunk 4]
```diff
@@ -30,11 +30,12 @@
...
+- name: Destroy deployment
+  register: ghpc_destroy
+  changed_when: ghpc_destroy.changed
+  ignore_errors: true
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml:36]: Same as above, it hasn't given me errors in testing, and it seems to allow any follow up tasks after this one to continue running.  I'm guessing it assumes that if there are any logging tasks after destruction that they could continue running.

File: tools/cloud-build/daily-tests/tests/hcls.yml [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+---
+
+test_name: hcls
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply tools/cloud-build/daily-tests/tests/hcls.yml:17]: Done

File: tools/cloud-build/daily-tests/ansible_playbooks/slurm-integration-test.yml [hunk 6]
```diff
@@ -146,19 +144,17 @@
...
-
+        terraform_apply_stderr_one_line: "{{ deployment.stderr }}"
     - name: Gather logs
       ansible.builtin.include_tasks:
         file: tasks/gather_startup_script_logs.yml
         apply:
           delegate_to: localhost
-
-    - name: Include rescue from terraform failure
+    - name: Include rescue from ghpc failure
```
[the remark this answers is not in the document]
Comment (cdunbar13) [reply tools/cloud-build/daily-tests/ansible_playbooks/slurm-integration-test.yml:155]: Done.

File: tools/cloud-build/daily-tests/ansible_playbooks/slurm-integration-test.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/slurm-integration-test.yml:91]: I am not 100% either but, since you what have is explicit and clear I am fine to leave it. I guess you could have 
```
      ansible.builtin.assert:	     
        that:	       
        - login_ip is defined	
        - login_ip != ""	
```

Do as you please and close this comment.

File: tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml:1]: Done

File: tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml:37]: Let's not add this ignore_errors unless there is a good specific reason to do so. The control flow of the tests is complicated and I do not want to alter it without reason.

File: tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (nick-stroud) [reply tools/cloud-build/daily-tests/ansible_playbooks/tasks/rescue_ghpc_failure.yml:35]: SG. If we think it is unused then cleaning it up is not a bad idea, but... I would do it in a separate PR to isolate the change and make it easy to identify and roll back if necessary. Your choice if you want to pursue.

**Review by nick-stroud — CHANGES_REQUESTED** [review]

**Review by nick-stroud — APPROVED** [review]
